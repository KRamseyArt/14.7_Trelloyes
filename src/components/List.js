import React from 'react';
import Card from './Card';
import './List.css';

export default function List(props) {
  return (
    <section className="List">
      <header>
        {props.header}
      </header>
      <div className="List-cards">
        {props.cards.length === 0 ? "empty" : props.cards.map((card, i) => {
          return (
            <Card
              key={i}
              id={card.id}
              handleDeleteCard={props.handleDeleteCard}
              title={card.title}
              content={card.content}
            />
          )
        })}
        <button
          type="button"
          onClick={
            () => props.handleAddRandomCard(props.id)}
          className="List-add-button"
        >
          + Add Random Card
        </button>
      </div>
    </section>
  )
}

List.defaultProps = {
  cards:[]
}