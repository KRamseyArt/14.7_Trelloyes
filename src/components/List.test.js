import React from 'react';
import ReactDOM from 'react-dom';
import List from './List';
import renderer from 'react-test-renderer';

describe('List component', () =>{
  it('renders without crashing', () =>{
    const div = document.createElement('div');
    ReactDOM.render(<List />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () =>{
    const tree = renderer
      .create(<List header="List" cards={[]}/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  })

  it('renders the UI as expected', () =>{
    const tree = renderer
      .create(<List header="List" cards={[{title:"title1", content:"content1"}, {title: 'title2', content: 'content2'}]}/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  })

  it('renders the UI without title and content', () =>{
    const tree = renderer
      .create(<List />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  })
})