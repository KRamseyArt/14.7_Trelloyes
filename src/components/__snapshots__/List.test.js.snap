// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component renders the UI as expected 1`] = `
<section
  className="List"
>
  <header>
    List
  </header>
  <div
    className="List-cards"
  >
    empty
    <button
      className="List-add-button"
      type="button"
    >
      + Add Random Card
    </button>
  </div>
</section>
`;

exports[`List component renders the UI as expected 2`] = `
<section
  className="List"
>
  <header>
    List
  </header>
  <div
    className="List-cards"
  >
    <div
      className="Card"
    >
      <button
        type="button"
      >
        Delete
      </button>
      <h3>
        title1
      </h3>
      <p>
        content1
      </p>
    </div>
    <div
      className="Card"
    >
      <button
        type="button"
      >
        Delete
      </button>
      <h3>
        title2
      </h3>
      <p>
        content2
      </p>
    </div>
    <button
      className="List-add-button"
      type="button"
    >
      + Add Random Card
    </button>
  </div>
</section>
`;

exports[`List component renders the UI without title and content 1`] = `
<section
  className="List"
>
  <header />
  <div
    className="List-cards"
  >
    empty
    <button
      className="List-add-button"
      type="button"
    >
      + Add Random Card
    </button>
  </div>
</section>
`;
